docs: add support guidelines (#40)
autowarefoundation/autoware-documentation: docs/help/support-guidelines.md

# Support guidelines

This page explains the support mechanisms we provide.

!!! warning

>     Before asking any help, search and read this documentation site carefully.

Reviewer: ```suggestion
    Before asking for help, please search and read this documentation site carefully.
```

Author: The word `please` is not recommended. So can you remove it?
https://developers.google.com/style/tone?hl=en#politeness-and-use-of-please

Reviewer: Wow that is a great site, I will check the recommendations there before commenting here in the future, thanks!

Also updated.

Author: I will write documentation guidelines for Autoware in the near future!

>     Also, understand and follow our [discussion guidelines](../contributing/discussion-guidelines/index.md) during your discussions.

Reviewer: ```suggestion
    Also, please follow the [discussion guidelines](../contributing/discussion-guidelines/index.md) for the discussions.
```

Reviewer: @kenji-miyake https://developers.google.com/style/person?hl=en It's described in the last part paragraph here.

Author: Great, thank you!
I might have to update other parts.

Author: Will fix them later in #51.

Reviewer: You can apply the changes here too.

Author: I'm not sure if I need `the` for `for the discussions.`
I'd like to wait for a comment from @LalithVipulananthan here, too.

Reviewer: @kenji-miyake i removed the last the.

Choose appropriate resources depending on what kind of help you need:

> - Documentation sites
>   - Various information
> - GitHub Discussions
>   - Questions
>   - Unconfirmed bugs
>   - Feature requests
>   - Design discussions
> - GitHub Issues
>   - Confirmed bugs
> - Slack
>   - Instant messaging between contributors
> - ROS Discourse
>   - General topics that should be widely announced

Reviewer: ```suggestion
- **Documentation sites**
  - Various information
- **GitHub Discussions**
  - Questions
  - Unconfirmed bugs
  - Feature requests
  - Design discussions
- **GitHub Issues**
  - Confirmed bugs
- **Slack**
  - Instant messaging between contributors
- **ROS Discourse**
  - General topics that should be widely announced
```

Just making the help sources bold.

Author: Before
![image](https://user-images.githubusercontent.com/31987104/157447555-6f6c2ae6-46c3-4d16-b184-dc56903e3696.png)

After
![image](https://user-images.githubusercontent.com/31987104/157447533-468e8c4f-f9d7-48e7-aae2-45de451c877c.png)

> The following sections are detailed descriptions of each resource.

Reviewer: ```suggestion
Following sections contain detailed descriptions of each these resources.
```

Since it is the first time we are referring to `Following sections` I dropped the `the` before it.

Author: Grammarly raises warnings. :thinking: 

![image](https://user-images.githubusercontent.com/31987104/157448159-087eff73-f2ad-4c3d-80f6-502df71b2470.png)

![image](https://user-images.githubusercontent.com/31987104/157448182-345f75b1-f889-4e01-a167-8185b5cbe29b.png)

Reviewer: https://www.grammarly.com/blog/articles-with-plural-nouns/

> Plural nouns can take either a definite article or no article at all. I think the is unnecessary there.

Reviewer 2: Grammatically speaking, it should be `The following`. However, I would argue that the line can be removed entirely, since it is clear from context that the following sections provide detail on the various communication channels available for support.

Author: Hmm, I see, thank you.

But I want people not to stop here and proceed to the following sections.
So how about changing here like the following?

```
Confirm the detailed descriptions of each resource in the following sections.
```

or,

```
```
See the following sections for the detailed descriptions of each resource.
```

Reviewer 2: Hmm. I see your point, but at the same time I think most people would naturally continue reading down the page. 

If we added links to the detailed description section for each communication channel in the list, would that be a clear enough indication that there is more detail below? 🤔

Author: @LalithVipulananthan How about this? 5fcad549196a5cb9a7c1392b84746a7099545b86

Reviewer 2: Excellent! 👍🏾

Author: Actually, I tried this first but didn't look good because there was little content. :sweat_smile: 
But now it looks good!

Author: @xmfcx The bold styles were removed because links are added instead. Is it okay for you?
If so, please resolve the conversation! :pray: 

![image](https://user-images.githubusercontent.com/31987104/157783213-8bc94431-565a-4714-96c7-e8fd2fcbf802.png)

## Documentation sites

[Docs guide](docs-guide.md) shows the list of useful documentation sites.
Visit them and see if there is any information related to your problem.

> Note that the documentation sites aren't always latest and perfect.

Reviewer: ```suggestion
Note that the documentation sites aren't always up-to-date and perfect.
```

> If you find some information is wrong, unclear, or missing in Autoware docs, feel free to submit a pull request following our [contribution guidelines](../contributing/index.md).

Reviewer: ```suggestion
If you find out that some information is wrong, unclear, or missing in Autoware docs, feel free to submit a pull request following our [contribution guidelines](../contributing/index.md).
```

!!!warning

    Since this documentation site is still under construction, there are some empty pages.

## GitHub Discussions

> If you have a problem with Autoware, check existing issues and questions and search similar issues first.

Reviewer: ```suggestion
If you encounter a problem with Autoware, please check existing issues and questions and search for similar issues first.
```

Author: Please remove all `please`. :pray:

Reviewer: Done.

- [Issues](https://github.com/autowarefoundation/autoware/issues)

  Note that Autoware has multiple repositories listed in [autoware.repos](https://github.com/autowarefoundation/autoware/blob/main/autoware.repos).  
  It is recommended to search across the repositories.

- [Questions](https://github.com/autowarefoundation/autoware/discussions/categories/q-a)

If no answer was found, create a new question thread [here](https://github.com/autowarefoundation/autoware/discussions/categories/q-a).

> If your question is not answered in a timely manner (about a week), mention maintainers and remind them.

Reviewer 3: I think this paragraph should be indented.

Author: Thank you, but it's not for `Questions`.
I meant that "if no answer was found in Issues or Questions".

Reviewer 4: ROS Answers is not included in these guidelines. Is this deliberate or should this option be available (in addition to Github Discussions)?

Reviewer 4: That makes sense. Thank you for clarifying.

> If your question is not answered in a timely manner (about a week), mention maintainers and remind them.

Reviewer: ```suggestion
If your question is not answered in a timely manner (about a week), please mention the maintainers to remind them.
```

Author: Hmm, could you please describe the difference?

Reviewer: Removed `please`, They remind the maintainers by mentioning them so I put `to` there instead of `and`. But I think both work. We can keep `and`

Reviewer 2: The line says to "mention the maintainers", but which maintainers specifically (does it depend on the specific repo, or anything like that?).

Author: I meant general `Autoware community's maintainers`, but no specific person from the maintainers.

Author: When do I have to add `then` for `if` sentences? :thinking:

Reviewer 2: @kenji-miyake That is a good question! It's a bit like using `なら` I think? 

I re-read that sentence to myself a few times (and looked for other examples online), and I think the `then` can be removed in this particular example to improve the flow of the sentence (apologies, as I saw that you had already made a commit for this earlier).

Author: No problem. Since Google's style guide also uses `then`, I think there is no problem.
https://developers.google.com/style

But later we can consider making the style consistent if necessary! :+1:

Also, there are other discussion types such as [feature requests](https://github.com/autowarefoundation/autoware/discussions/categories/feature-requests) or [design discussions](https://github.com/autowarefoundation/autoware/discussions/categories/design).
Feel free to open or join such discussions.

## GitHub Issues

If you have a problem and you have confirmed it is a bug, find the appropriate repository and create a new issue there.

> If you can't determine the appropriate repository, you should not create an issue yet.

Reviewer: ```suggestion
If you can't determine the appropriate repository, please ask about where to create it in the [Q&A section](https://github.com/autowarefoundation/autoware/discussions/categories/q-a). And the maintainers will help you decide.
```

Author: - Please remove `please`.
- `Q&A section` -> `Q&A category` is better?
- I'm not sure whether I should add `the` for `maintainers`.
    - Will follow the review result of https://github.com/autowarefoundation/autoware-documentation/pull/40/files/ae7fe87953f66dff58bde36fdb1568d82972ef43#r822744588.

Reviewer: `category` is better.

removed `please`.

Author: Also, I wonder if I can use `can't` or `don't` instead of `cannot` or `do not` for technical documents.
Seeing https://developers.google.com/style?hl=en, it's allowed?

But I'd like to hear @LalithVipulananthan's comment to clarify the rule.

Reviewer: You can of course use, no ome can argue against that.

Reviewer 2: I have to admit that I am not sure about what is acceptable for a modern technical guide with respect to `can't` or `don't` instead of `cannot` or `do not`, so would defer to the Google developer documentation style guide in this case.

My alternative suggestion for the line above:

```suggestion
If you can't determine the appropriate repository, ask the maintainers for help by creating a new Discussion in the [Q&A category](https://github.com/autowarefoundation/autoware/discussions/categories/q-a).
```

Author: Thank you!
I don't stick to small differences, but could I ask some questions so that I can properly review other developers in the future?

- Should I add `the` for `maintainers` here?
- Why did you capitalize `Discussion`?

Reviewer 2: - In the sentence above, it should be `ask the maintainers` rather than `ask maintainers`. Does that answer your question, or was there a different example that you wanted me to clarify?
- For `Discussion`, I capitalized it because the feature is referred to as [GitHub Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). Then I read that documentation and noticed that they don't capitalize the word when referring to individual discussions so it should be `discussion` instead.

!!!warning

    Do not create issues for questions or unconfirmed bugs. If such issues are created, maintainers will transfer them to GitHub Discussions.

> If you want to fix the bug by yourself, discuss the approach with maintainers and submit a pull request.

Reviewer 3: In this case, do I need to create an Issue? Or can I create a PR directly?

Author: It's written here.
https://github.com/autowarefoundation/autoware-documentation/pull/43/files#diff-1aeecae8f389487f8c42feed5ae17f247adf21aea14e59918c3b6ac7f8d1b033R54

You don't have to create an issue for small fixes.

Author: However, since this section is for bugs, I guess an issue is already created. :thinking:

Reviewer 3: Thank you. I considered the case that the bug is found and fixed quickly, but this should be a small PR.

Author: You can add some examples here.
https://github.com/autowarefoundation/autoware-documentation/pull/43/files#diff-1aeecae8f389487f8c42feed5ae17f247adf21aea14e59918c3b6ac7f8d1b033R58-R61

Reviewer 3: Resolved.

## Slack

> Autoware has a Slack workspace for casual communication between contributors.

Author: Yes, but now it is not `autoware-auto` anymore. So I think we need another channel.

> Autoware has a Slack workspace for casual communication between contributors.

Author: By the way, I feel like it excludes newbies from the word `committers`.
So I personally prefer `contributors-autoware`, or just `autoware` looks good to me.

> Autoware has a Slack workspace for casual communication between contributors.

Author: Also, I believe discussions between "committers" should be done on GitHub because not all developers join Slack.
They can communicate on Slack, but they should write the conclusion on GitHub as well.
I mean, decision makings shouldn't be closed inside Slack.

It's so important for the community.

> Autoware has a Slack workspace for casual communication between contributors.

Reviewer 4: You raise some good points. The committers Slack channel is currently used by members of the ASWG. At the next TSC there will be a discussion about the Slack vs. Discord proposal from Fatih. We can wait to hear the outcome of that proposal before making changes to the Slack channels.

> Autoware has a Slack workspace for casual communication between contributors.

Author: I see, thank you!

> At the next TSC there will be a discussion about the Slack vs. Discord proposal from Fatih.

@BonoloAWF cc @xmfcx @mitsudome-r I have one thing to ask you to consider there.
TIER IV is thinking of connecting the Slack workspace between our company and the Autoware Developer's Community to make more collaborations.
In that case, I think it would be a great benefit for the community because we believe we can provide quick responses to the posts.

So, I'd appreciate it if you could take that into consideration and decide our communication platform. :pray:

> Autoware has a Slack workspace for casual communication between contributors.

Reviewer: @kenji-miyake I think it's already been decided, we were just waiting for the right moment to do it (do it with transition to github).
And the right way to announce it to the community.
I've started the ground-work for the migration.

Maybe you can set up a discord or slack bot to connect it to your platform but I suggest you switch to the discord as a company too, we've done it in the past and everyone is happy about it.

Here is the slide I've shared in the TSC call: https://docs.google.com/presentation/d/1hHyD-cWCOoO7hUfhm4ws5eL68HoRT2MzNYZSy-VVsL4/edit?usp=sharing

> Autoware has a Slack workspace for casual communication between contributors.

Author: @xmfcx Oh, really? If it's the decision by TSC, it's okay for me.
However, please note that we should consider "fairness of information" as said above.
For example, we should take care not to close discussions only within voice channels.

> I suggest you switch to the discord as a company too

Unfortunately, it has a hurdle. But I'll try it as much as possible for the community.

> Autoware has a Slack workspace for casual communication between contributors.

Reviewer: > However, please note that we should consider "fairness of information" as said above.
> For example, we should take care not to close discussions only within voice channels.

Don't worry about that, any decision made will end up in an issue before it is committed as an end result.

> @xmfcx Oh, really? If it's the decision by TSC, it's okay for me.

Mainly no one opposed and many participants supported the move, including with the multiple account requirement for each slack server too as a reason.

Discord will lower the barrier of entry for the new comers for this reason too. They don't need to re-register just for autoware.

> Autoware has a Slack workspace for casual communication between contributors.

Author: @xmfcx Will you update this to `Discord` now? Or at a later date?

> Autoware has a Slack workspace for casual communication between contributors.

Reviewer: @kenji-miyake we can keep it like this for now, I will create a PR for it later on.

> Autoware has a Slack workspace for casual communication between contributors.

Author: Okay, thank you!

> Autoware has a Slack workspace for casual communication between contributors.

Author: https://github.com/autowarefoundation/autoware-documentation/issues/7#issuecomment-1062895769

For example, this is a good place for the following activities:

- Introduce yourself to the community.
- Chat with contributors.
- Take a quick straw poll.

Note that it is not the right place to get help for your issues.

## ROS Discourse

> If you want to widely discuss a topic with the general Autoware community or ask a question not related to Autoware's bugs, post to [the Autoware category on ROS Discourse](https://discourse.ros.org/c/autoware).

Reviewer: ```suggestion
If you want to widely discuss a topic with the general Autoware and ROS community or ask a question not related to Autoware's bugs, please post to [the Autoware category on ROS Discourse](https://discourse.ros.org/c/autoware).
```

Author: I guess we don't use the Autoware category for ROS community.

Reviewer: But anyone in the ROS community will be able to see and join to the discussion there.

Author: Hmm, I think it should be added here.
https://github.com/autowarefoundation/autoware-documentation/blob/main/docs/help/docs-guide.md

But if you strongly want to add it here, I will apply after you remove `please`.

Reviewer: I think you can add it here, removed `please`.

Author: Okay!
